code/datums/immutable_list.dm

```diff
@@ -0,0 +1,137 @@
+///Real globals have no datum access overhead, so we do both to maintain GLOB having all the important bits.
+GLOBAL_REAL(_immutable_list_repo, /list) = list()
+GLOBAL_LIST_INIT(immutable_list_repo, global._immutable_list_repo)
```

Comment:
Why is this worth it

Reply:
maximum speed, i dont think a list standin should be sacrificing any speed where not necessary.

Reply:
It is nice that you feel that way but how much is "maximum" and why is it worth spawning readability from satan here?

Reply:
its on 1-2 less datum accesses per mutation/creation of an immutable list

Reply:
this doesn't really answer my question. How hot is this code? and why does it warrant it?

Reply:
not incredibly hot right now, but if this is used more it will scale very quickly. Built for scalabilty.